frontend/packages/helm-plugin/src/components/forms/HelmChartRepository/__tests__/helmchartrepository-create-utils.spec.ts

```diff
@@ -44,4 +45,35 @@ describe('HelmChartRepository create utils', () => {
     defaultResource = getDefaultResource('test-ns', referenceForModel(HelmChartRepositoryModel));
     expect(defaultResource).toEqual(defaultHCR);
   });
+  it('should include basicAuthConfig only for ProjectHelmChartRepository', () => {
+    // Test ProjectHelmChartRepository includes basicAuthConfig
+    const projectFormData = _.cloneDeep(sampleHelmChartRepositoryFormData);
+    projectFormData.scope = 'ProjectHelmChartRepository';
+    projectFormData.basicAuthConfig = 'test-basic-auth-secret';
+    const projectRepo = convertToHelmChartRepository(projectFormData, 'test-ns');
+    expect(projectRepo.spec.connectionConfig.basicAuthConfig).toEqual({
+      name: 'test-basic-auth-secret',
+    });
+
+    // Test cluster-scoped HelmChartRepository excludes basicAuthConfig
+    const clusterFormData = _.cloneDeep(sampleHelmChartRepositoryFormData);
+    clusterFormData.scope = 'HelmChartRepository';
+    clusterFormData.basicAuthConfig = 'test-basic-auth-secret';
+    const clusterRepo = convertToHelmChartRepository(clusterFormData, 'test-ns');
+    expect(clusterRepo.spec.connectionConfig.basicAuthConfig).toBeUndefined();
+  });
+  it('should convert form data with basicAuthConfig correctly', () => {
+    const repoWithBasicAuth: HelmChartRepositoryType = {
+      ...sampleProjectHelmChartRepository,
+      spec: {
+        ...sampleProjectHelmChartRepository.spec,
+        connectionConfig: {
+          ...sampleProjectHelmChartRepository.spec.connectionConfig,
+          basicAuthConfig: { name: 'my-basic-auth' },
+        },
+      },
+    };
```

Comment:
Wouldn't it be sufficient and simpler to use the same idiom you used in the previous tests:

```suggestion
    const repoWithBasicAuth = _.cloneDeep(sampleProjectHelmChartRepository);
    repoWithBasicAuth.spec.connectionConfig.basicAuthConfig = { name: 'my-basic-auth' };
```